src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/alignment/AssemblyContigAlignmentsConfigPicker.java

```diff
@@ -520,92 +684,71 @@ static GoodAndBadMappings splitGaps(final GoodAndBadMappings configuration) {
-     * and be marked as having an SV on the corresponding location on the canonical chromosome.
-     *
-     * An assembly contig from this sample may have two equally well scored alignment configurations, where
-     *  one configuration has split alignments to canonical chromosomes, hence indicating the SV, whereas
-     *  the other configuration has a single, often very good (or even better) alignment to a non-canonical chromosome.
-     * We send down the chimeric alignment configuration for inference, but notes down that a non-canonical chromosome
-     * in the reference input could have already captured the SV on this sample.
-     *
-     * @return  {@code null} if the non-canonical chromosome mapping doesn't offer a better score,
-     *          otherwise the non-canonical chromosome mapping
-     */
     @VisibleForTesting
-    static AlignmentInterval getBetterNonCanonicalMapping(final Set<String> canonicalChromosomes,
-                                                          final List<AlignmentInterval> goodMappings,
-                                                          final int maxCanonicalChrAlignerScore) {
-        final List<AlignmentInterval> canonicalMappings = new ArrayList<>(goodMappings.size());
-        final List<AlignmentInterval> nonCanonicalMapping = new ArrayList<>();
-        for (final AlignmentInterval alignment : goodMappings) {
-            if ( canonicalChromosomes.contains(alignment.referenceSpan.getContig()) )
-                canonicalMappings.add(alignment);
-            else
-                nonCanonicalMapping.add(alignment);
-        }
-        if ( canonicalMappings.isEmpty() )
-            return null;
-        if ( nonCanonicalMapping.size() == 1 &&
-                (canonicalMappings.size() > 1 || canonicalMappings.get(0).containsGapOfEqualOrLargerSize(GAPPED_ALIGNMENT_BREAK_DEFAULT_SENSITIVITY)) ) {
-            final double canonicalScore = computeScoreOfConfiguration(canonicalMappings, canonicalChromosomes, maxCanonicalChrAlignerScore);
-            final double nonCanonicalScore = computeScoreOfConfiguration(nonCanonicalMapping, canonicalChromosomes, maxCanonicalChrAlignerScore);
-            return ( canonicalScore > nonCanonicalScore ) ? null : nonCanonicalMapping.get(0);
+    static boolean gappedAlignmentOffersBetterCoverage(final AlignmentInterval gapped,
+                                                       final AlignmentInterval overlappingNonGapped) {
+        final int diff = gapped.getSizeOnRead() - overlappingNonGapped.getSizeOnRead();
+        if ( diff == 0) {
+            return gapped.alnScore > overlappingNonGapped.alnScore;
         } else {
-            return null;
+            return diff > 0;
         }
     }
 
-    //==================================================================================================================
+    // functionality group 3.2: after gap split, further reclassify some good mapping as bad
```

Comment:
I'd add a little description of which mappings are getting filtered out here to the comment.